### ISSUE TITLE
Import Wallet/SEED shall we add a scan QR code function?

### ISSUE THREAD

**Issue by zhangzhongnan928** [AlphaWallet/alpha-wallet-android#811]
There are many wallets provide Showing seed as QR Code
![IMAGE 2019-08-15 23:51:04](https://user-images.githubusercontent.com/33795543/63099044-8d2d7e00-bfb7-11e9-85c4-f860d457c407.jpg)

We should provide a scan QR code function for user to easily import a wallet.

**Comment by JamesSmartCell** [comment]
Yes this would be useful, also export private key as QR and import private key as QR same as metamask. Then you can seamlessly switch between the two which is helpful for developers.

**Comment by hboon** [comment]
We briefly discussed about this. But I don't think we arrived at a conclusion. There's the concern that this encourages bad security practice because, there's a good chance the user generates the QR code using a website.

I'm fine with doing it. No strong opinions here. @colourful-land ?

But if we decide to do it, let's do it after this sprint, like https://github.com/AlphaWallet/alpha-wallet-android/issues/780

**Comment by hboon** [comment]
- [ ] Decide if we do it. @colourful-land 
- [ ] If we decide to do it, do it after the sprint (or someone can disagree)

**Comment by zhangzhongnan928** [comment]
Don't provide export into QR code(this is super bad, it encourage the user to save the seeds in a photo album or cloud without encryption).

Only prove import from scanning QR Codes which are generated from other wallets.

**Comment by James-Sangalli** [comment]
Would argue against it because of security concerns and lack of adoption for such a method with seeds.

**Comment by zhangzhongnan928** [comment]
> Would argue against it because of security concerns and lack of adoption for such a method with seeds.

You against “export to QR” or “ import from QR”
I don’t see any security concerns about “ import from QR.

> Don't provide export into QR code(this is super bad, it encourage the user to save the seeds in a photo album or cloud without encryption).
> 
> Only prove import from scanning QR Codes which are generated from other wallets.

**Comment by hboon** [comment]
Bump @colourful-land until we settle on a different arbitrator

**Comment by jzaki** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [[__] Victor: Import Wallet/SEED shall we add a scan QR code function?](https://trello.com/c/b47sz01Q/31-victor-import-wallet-seed-shall-we-add-a-scan-qr-code-function)

**Comment by jzaki** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [[700] As a wallet user, I want to export/import my wallet's SEED phrase more readily, so that I'm not frustrated with typing long seed phrases](https://trello.com/c/b47sz01Q/31-700-as-a-wallet-user-i-want-to-export-import-my-wallets-seed-phrase-more-readily-so-that-im-not-frustrated-with-typing-long-seed)
